bigbluebutton-html5/imports/ui/components/whiteboard/annotations/triangle/component.jsx

```diff
@@ -1,52 +1,64 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import AnnotationHelpers from '../helpers.js';
+import AnnotationHelpers from '../helpers';
 
-export default class TriangleDrawComponent extends React.Component {
-  constructor(props) {
-    super(props);
+export default class TriangleDrawComponent extends Component {
+
+  shouldComponentUpdate(nextProps) {
+    return this.props.version !== nextProps.version;
   }
 
   getCoordinates() {
-    let path = '';
-
-    // points[0] and points[1] are x and y coordinates of the top left corner of the annotation obj
-    // points[2] and points[3] are x and y coordinates of the bottom right corner of the annotation obj
-    const xBottomLeft = this.props.annotation.points[0];
-    const yBottomLeft = this.props.annotation.points[3];
-    const xBottomRight = this.props.annotation.points[2];
-    const yBottomRight = this.props.annotation.points[3];
+    const { slideWidth, slideHeight } = this.props;
+    const { points } = this.props.annotation;
+
+    // points[0] and points[1] are x and y coordinates of the top left corner of the annotation
+    // points[2] and points[3] are x and y coordinates of the bottom right corner of the annotation
+    const xBottomLeft = points[0];
+    const yBottomLeft = points[3];
```

Comment:
this should probably be `points[1]`?

Reply:
Those coordinates are correct if you're looking for the bottom-left corner. The points given are top-left and bottom-right so you have to pick from both if you want a different corner.

Reply:
`points[3]` is both `yBottomRight` and `yBottomLeft`?

Reply:
points[3] on it's own is the bottom of the shape

Reply:
Yes, so points[0] and points[1] are top-left X and Y, and points[2] and points[3] are bottom-right X and Y.
As a result bottom left takes X from the top-left ( [0] )and Y from bottom-right ( [3] ).
And top-right takes [2] and [1] accordingly.